src/Engine/Graphics.cpp

```
@@ -456,26 +487,305 @@ namespace {
+			// validate required device extensions
+			std::vector<const char*> required_device_extensions(std::begin(device_extensions), std::end(device_extensions));
+
+			for (auto [extensionName, specVersion] : extensions)
+			{
+				l->debug(std::format("Device extension name: {}", extensionName));
+				for (const char* extension_name : device_extensions)
+				{
+					if (strcmp(extension_name, extensionName) == 0)
+					{
+						l->debug(std::format("Requiring device extension: {}", extension_name));
+						device_extensions.push_back(extension_name);
```

cppcheck (2.10-2)

[error] Calling 'push_back' while iterating the container is invalid.

(invalidContainerLoop)